mitiq/cdr/cdr_execution.py

```diff
@@ -0,0 +1,202 @@
+    circuit: Circuit,
+    executor: Callable[[Circuit], dict],
+    simulator: Callable[[Circuit], Union[dict, np.ndarray]],
+    observables: List[np.ndarray],
+    num_training_circuits: int,
+    fraction_non_clifford: float,
+    ansatz: Callable[[np.ndarray, List], List] = linear_fit_function,
+    num_parameters: int = None,
+    scale_noise: Optional[Callable[[Circuit, float], Circuit]] = None,
+    scale_factors: Optional[List[float]] = None,
+    **kwargs: dict,
+) -> (List[List], List[List]):
+    """Function for the calculation of an observable from some circuit of
+    interest to be mitigated with CDR (or vnCDR) based on [Czarnik2020]_ and
+    [Lowe2020]_.
+
+    The circuit of interest must be compiled in the native basis of the IBM
+    quantum computers, that is {Rz, sqrt(X), CNOT}, or such that all the
+    non-Clifford gates are contained in the Rz rotations.
+
+    The observable/s to be calculated should be input as an array or a list of
+    arrays representing the diagonal of the observables to be measured. Note
+    these observables MUST be diagonal in z-basis measurements corresponding to
+    the circuit of interest.
+
+    Returns list of raw observables (at many noise levels) and mitigated
+    observables.
+
+    This function returns the mitigated observable/s.
+    Args:
+        circuit: circuit of interest compiled in the correct basis.
+        executor: user defined function taking a cirq Circuit object and
+                  returning a dictionary of counts.
+        simulator: user defined function taking a cirq Circuit object and
+                   returning either a simulated dictionary of counts or an
+                   np.ndarray representing the state vector.
+        observable: list of arrays containing the diagonal elements of
+                    observable/s of interest to be mitigated. If a list is
+                    passed all these observables will be mitigates with the
+                    same training set.
+        num_training_circuits: number of training circuits to be used in the
+                               mitigation.
+        fraction_non_clifford: the fraction of non-Clifford gates to be
+                               subsituted in the training circuits. The higher
+                               this fraction the more costly the simulations,
+                               but more successful the mitigation.
+        ansatz: the function to map noisy to exact data. Takes array of noisy
+                and data and parameters returning a float.
+        num_parameters: the number of paramters the ansatz takes.
+        scale_noise: optional argument containing a user defined function on
+                     how to increase the noise. If this argument is given then
+                     the mitigation method will be vnCDR.
+        scale_factors: factors by which to scale the noise, should not
+                               include 1 as this is just the original circuit.
```

Comment:
```suggestion
        scale_factors: factors by which to scale the noise, should not
                               include 1 as this is just the original circuit.
```
It would be much better if scale_factors can include 1 because this is the convention that we use everywhere in Mitiq. If necessary, you can always internally remove all the 1 elements (usually the just first).